src/test/java/com/amazon/dtasdk/utils/BinaryUtilsTest.java

```diff
@@ -0,0 +1,44 @@
+package com.amazon.dtasdk.utils;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class BinaryUtilsTest {
+
+    @Rule public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void testFromHex() {
+        Assert.assertArrayEquals(new byte[] { 0x7f }, BinaryUtils.fromHex("7f"));
+        Assert.assertArrayEquals(new byte[] { 0x1a, 0x2b, 0x3c, 0x4d }, BinaryUtils.fromHex("1a2b3c4d"));
+        Assert.assertArrayEquals(new byte[] {}, BinaryUtils.fromHex(""));
+
+        thrown.expect(StringIndexOutOfBoundsException.class);
+        BinaryUtils.fromHex("0f0");
+    }
+
+    @Test
+    public void testToBase64() throws Exception {
+        Assert.assertEquals("dGVzdCBzdHJpbmc=", BinaryUtils.toBase64("test string".getBytes()));
+        Assert.assertEquals("", BinaryUtils.toBase64(new byte[] {}));
+        Assert.assertEquals("AQIDBA==", BinaryUtils.toBase64(new byte[] {0x01, 0x02, 0x03, 0x04}));
+    }
+
+    @Test
+    public void testFromBase64() throws Exception {
```

Comment:
Is there a way we can reach the unsupportedEncodingException?

Reply:
Presumably pass in a base64 string that is encoded psychopathically and that'll do the trick.